chatsky/llm/prompt.py

```diff
@@ -40,12 +43,179 @@ class Prompt(BaseModel):
+    
+    This class implements BasePrompt interface and provides concrete implementation
+    for converting various message types (string, Message, BaseResponse) into a Message object.
+    """
+
+    message: AnyResponse
+    """
+    The message content of the prompt. Can be:
+    - string: will be converted to Message with text field
+    - Message: will be used as is
+    - BaseResponse: will be evaluated in the context to get Message
+    """
+
     def __init__(self, message: Union[MessageInitTypes, BaseResponse], position: Optional[float] = None):
+        """
+        Initialize a new Prompt.
+
+        :param message: The message content for this prompt
+        :param position: Optional position in the context sequence
+        """
         super().__init__(message=message, position=position)
 
     @model_validator(mode="before")
     @classmethod
     def validate_from_message(cls, data):
+        """
+        Validate and convert input data to proper format.
+        Allows creating Prompt from string, Message or BaseResponse directly.
+
+        :param data: Input data to validate
+        :return: Validated data in proper format
+        """
         if isinstance(data, (str, Message, BaseResponse)):
             return {"message": data}
         return data
+
+    async def message(self, ctx: Context) -> Message:
+        """
+        Get the message content for this prompt.
+        Implements the abstract method from BasePrompt.
+
+        :param ctx: Current dialog context
+        :return: Message object containing the prompt content
+        :raises ValueError: If message cannot be converted to proper format
+        """
+        try:
+            if isinstance(self.message, Message):
+                return self.message
+            elif isinstance(self.message, BaseResponse):
+                return await self.message(ctx)
+            else:
+                return Message(text=str(self.message))
+        except Exception as e:
+            raise ValueError(f"Failed to get message content: {str(e)}")
+
+
+class FewShotExamplePrompt(BasePrompt):
+    """
+    Prompt class that supports few-shot examples with templates.
+    Uses Langchain's example selectors and prompt templates for few-shot learning.
+    """
+
+    template: str = Field(
+        ...,
+        description="Template string that defines the structure of each example. Supports {variable} placeholders."
+    )
+
+    examples: List[Dict[str, Any]] = Field(
+        default_factory=list,
+        description="List of example dictionaries. Each must include values for all variables used in the template."
+    )
+
+    example_selector: Optional[Any] = Field(
+        None,
+        description="Optional LangChain example selector to dynamically choose examples at runtime."
+    )
+
+    prefix: Optional[str] = Field(
+        None,
+        description="Text to prepend before the examples (e.g., instruction or context)."
```

Comment:
It's better to set field descriptions as docstrings.
That way it is properly processed by all that need docstrings (e.g. documentation builders, IDE hints).